test/blackbox-tests/test-cases/melange/promote.t

```diff
@@ -0,0 +1,35 @@
+Test melange.emit promotion
```

Comment:
Would it make sense to add a case with a library? As it follows a bit different logic than melange.emit.

Reply:
added in [540d021](https://github.com/ocaml/dune/pull/6500/commits/540d021f6efda0ae3633511c9ca12212960fd00d)